src/site/content/en/fast/codelab-optimize-third-party-javascript/index.md

```diff
@@ -0,0 +1,248 @@
+
+```html
+<iframe width="560" height="315" data-src="https://www.youtube.com/embed/lS9D6w1GzGY" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+```
+
+`data-src` is a [data attribute](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes), which allows you to store extra information on standard HTML elements. A data attribute can be named anything, as long as it starts with "data-". 
+
+An iframe without a `src` simply won't load. 
+
+### Step 2: Use Intersection Observer to lazy-load the video
+
+To load the video when a user scrolls to it you need to know when that happens. That’s where the Intersection Observer API steps in. The Intersection Observer API lets you register a callback function that's executed whenever an element you want to track enters or exits the viewport.
+
+To get started, create a new file and name it `lazy-load.js`:
+
+{% Instruction 'create' %}
+
+Create a new `IntersectionObserver` and pass it a callback function to run:
+
+```js
+// create a new Intersection Observer
+let observer = new IntersectionObserver(callback);
+```
+
+Now give `observer` a target element to watch (the video iframe in this case) by passing it as an argument in the `observe` method:
+
+```js
+// the element that you want to watch
+const element = document.querySelector('iframe');
+
+// register the element with the observe method
+observer.observe(element);
+```
+
+`callback` receives a list of [`IntersectionObserverEntry`](https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserverEntry) objects and the [`IntersectionObserver`](https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserver) object itself. Each entry contains a [`target`](https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserverEntry/target) element and properties that describe its dimensions, position, the time it entered the viewport, and more. One of the properties of `IntersectionObserverEntry` is [`isIntersecting`](https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserverEntry/isIntersecting)—a boolean value that equals `true` when the element enters the viewport.
+
+In this example, the `target` is the `iframe`. The [`isIntersecting`](https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserverEntry/isIntersecting) equals `true` when `target` enters the viewport. To see this in action, replace `callback` with the following function: 
+
+```js/1-7/0
+let observer = new IntersectionObserver(callback);
+let observer = new IntersectionObserver(function(entries, observer) { 
+    entries.forEach(entry => {
+      console.log(entry.target);
+      console.log(entry.isIntersecting);
+    });
+  });
+```
+{% Instruction 'preview', 'ol' %}
+{% Instruction 'devtools-console', 'ol' %}
+
+Try scrolling up and down. You should see the value of `isIntersecting` change and the target element logged to the console.
```

Comment:
This isn't working for me.

Reply:
What, lazy loading? And in the final version that I've linked below?

Reply:
Tried again in a remixed Glitch, worked for me. If you can dig up a link to your remix, I can take a look to see if there's an issue.

Reply:
Just tried it again, but still not working: https://glitch.com/~hospitable-trigonometry